distributed/tests/test_scheduler.py

```diff
@@ -3118,20 +3114,12 @@ async def test_worker_heartbeat_after_cancel(c, s, *workers):
     computation where the worker returns metrics about duration, type, etc. and
     the scheduler doesn't handle the forgotten task gracefully.
 
-    Failures are not triggered reliably since the race conditions for this error
-    case are very hard to produce. Likelihood of failure increases with the
-    number of workers.
-
     See also https://github.com/dask/distributed/issues/4587
     """
-    da = pytest.importorskip("dask.array")
     for w in workers:
         w.periodic_callbacks["heartbeat"].stop()
-    x = da.random.random((2000000, 100), chunks=(10000, None))
-    svd = da.linalg.svd(x)
-
-    futs = c.compute(svd)
 
+    futs = c.map(slowinc, range(100), delay=0.1)
     while not s.tasks:
         await asyncio.sleep(0.001)
 
```

Comment:
Redundant with the following sleep

Reply:
Also the sleep after the cancel seems unnecessary; why don't you simply start bombarding the workers with heartbeat requests straight away?
```
    while s.tasks:
        await asyncio.sleep(0.001)

Reply:
```python
while sum(w.executing_count for w in workers) < len(workers) / 2:
```
Why don't you wait for all workers fullstop? (remove the / 2). slowinc should reliably fill the whole cluster.

Reply:
done